internal/resourceproviders/required.go

```diff
@@ -3,22 +3,131 @@
+// own set. They can also choose to not register any RPs, and manage these out-of-band, if they wish.
+//
+// Note that resource providers are case-sensitive.
+//
+// Official Docs: https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/azure-services-resource-providers
+
+type ResourceProviders map[string]struct{}
+
+func (r ResourceProviders) Add(providers ...string) {
+	for _, p := range providers {
+		r[p] = struct{}{}
+	}
+}
+
+func (r ResourceProviders) Merge(a ResourceProviders) ResourceProviders {
+	for k, v := range a {
+		r[k] = v
+	}
+	return r
+}
+
+// Core is a minimal set of RPs, and will be the default set in a future major version of the provider.
+func Core() ResourceProviders {
+	return ResourceProviders{
+		"Microsoft.Authorization":       {},
+		"Microsoft.Compute":             {},
+		"Microsoft.CostManagement":      {},
+		"Microsoft.ManagedIdentity":     {},
+		"Microsoft.MarketplaceOrdering": {},
+		"Microsoft.Network":             {},
+		"Microsoft.Resources":           {},
+		"Microsoft.Storage":             {},
+	}
+}
+
+// Extended is an expanded set of resource providers, as suggested by the community.
+func Extended() ResourceProviders {
+	return Core().Merge(ResourceProviders{
+		"Microsoft.ApiManagement":        {},
+		"Microsoft.AppConfiguration":     {},
+		"Microsoft.AppPlatform":          {},
+		"Microsoft.Automation":           {},
+		"Microsoft.Cache":                {},
+		"Microsoft.Cdn":                  {},
+		"Microsoft.ContainerInstance":    {},
+		"Microsoft.ContainerRegistry":    {},
+		"Microsoft.ContainerService":     {},
+		"Microsoft.DBforMariaDB":         {},
+		"Microsoft.DBforMySQL":           {},
+		"Microsoft.DBforPostgreSQL":      {},
+		"Microsoft.DataFactory":          {},
+		"Microsoft.DataLakeAnalytics":    {},
+		"Microsoft.DataLakeStore":        {},
+		"Microsoft.DataMigration":        {},
+		"Microsoft.DataProtection":       {},
+		"Microsoft.Databricks":           {},
+		"Microsoft.DevTestLab":           {},
+		"Microsoft.Devices":              {},
+		"Microsoft.DocumentDB":           {},
+		"Microsoft.EventGrid":            {},
+		"Microsoft.EventHub":             {},
+		"Microsoft.HDInsight":            {},
+		"microsoft.insights":             {},
+		"Microsoft.KeyVault":             {},
+		"Microsoft.Kusto":                {},
+		"Microsoft.Logic":                {},
+		"Microsoft.Maintenance":          {},
+		"Microsoft.Management":           {},
+		"Microsoft.NotificationHubs":     {},
+		"Microsoft.OperationalInsights":  {},
+		"Microsoft.OperationsManagement": {},
+		"Microsoft.PowerBIDedicated":     {},
+		"Microsoft.Relay":                {},
+		"Microsoft.Security":             {},
+		"Microsoft.SecurityInsights":     {},
+		"Microsoft.ServiceBus":           {},
+		"Microsoft.ServiceFabric":        {},
+		"Microsoft.SignalRService":       {},
+		"Microsoft.Sql":                  {},
```

Comment:
Same for this one 
```suggestion
```

Reply:
This is the same RP as the current generation so we'll keep this in